libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts

```diff
@@ -71,26 +74,91 @@ export class ModalWrapperComponent implements Modal, AfterViewInit, OnInit, OnDe
     private injector: Injector,
     private elementRef: ElementRef<HTMLElement>,
     private renderer: Renderer2,
-    private resizeObserverService: ResizeObserverService
-  ) {
-    this.observeViewportResize();
-  }
+    private resizeObserverService: ResizeObserverService,
+    private windowRef: WindowRef
+  ) {}
 
   ngOnInit(): void {
+    this.observeViewportResize();
     this.ionModalElement = this.elementRef.nativeElement.closest('ion-modal');
     this.listenForIonModalDidPresent();
     this.listenForIonModalWillDismiss();
     this.componentPropsInjector = Injector.create({
       providers: [{ provide: COMPONENT_PROPS, useValue: this.config.componentProps }],
       parent: this.injector,
     });
+
+    this.moveUnderlyingModalBackdropsInFrontOfModals();
+  }
+
+  /*
+   * This is done by TS and not by scss, because it otherwise will be delayed when reverting back the z-index on will-close
+   */
+  private moveUnderlyingModalBackdropsInFrontOfModals() {
+    if (!this.ionModalElement) return;
+    const container = this.ionModalElement.parentElement;
+    if (!container) return;
+
+    const underlyingModalElements = this.ionModalElement.parentElement.querySelectorAll(
+      'ion-modal:not(:last-of-type)'
+    );
+    if (underlyingModalElements.length === 0) return;
+
+    underlyingModalElements.forEach((modalElement) => {
+      this.renderer.setStyle(modalElement.querySelector('ion-backdrop'), 'z-index', '20');
+    });
+
+    this.willClose.then(() => {
+      underlyingModalElements.forEach((modalElement) => {
+        this.renderer.setStyle(modalElement.querySelector('ion-backdrop'), 'z-index', '2');
+      });
+    });
+  }
+
+  private getAvailableContentHeight(modalWrapper: any): number {
```

Comment:
Consider using strong typing:
```suggestion
  private getAvailableContentHeight(modalWrapper: HTMLElement): number {
```